api/controllers/document.js

```diff
@@ -457,13 +458,13 @@ exports.protectedOpen = function (args, res, next) {
 //  Create a new document
 exports.protectedPost = async function (args, res, next) {
   console.log("Creating new protected document object");
-  var project = args.swagger.params.project.value;
-  var _comment = args.swagger.params._comment.value;
-  var upfile = args.swagger.params.upfile.value;
-  var guid = intformat(generator.next(), 'dec');
-  var ext = mime.extension(args.swagger.params.upfile.value.mimetype);
-  var tempFilePath = uploadDir + guid + "." + ext;
   try {
+    var project = args.swagger.params.project.value;
+    var _comment = args.swagger.params._comment.value;
+    var upfile = args.swagger.params.upfile.value;
+    var guid = intformat(generator.next(), 'dec');
+    var ext = mime.extension(args.swagger.params.upfile.value.mimetype);
+    var tempFilePath = uploadDir + guid + "." + ext;
```

Comment:
It may well have been done to make this app more "containerizable." I can't take credit for this code though. I just moved it into a try catch block for better error handling